crates/bevy_color/src/color_ops.rs

```diff
@@ -0,0 +1,43 @@
+/// Methods for changing the luminance of a color. Note that these methods are not
+/// guaranteed to produce the same results in different color spaces, but are only
+/// approximations.
+pub trait Luminance: Sized {
+    /// Return the luminance of this color (0.0 - 1.0).
+    fn luminance(&self) -> f32;
+
+    /// Return a new version of this color with the given luminance.
+    fn with_luminance(&self, alpha: f32) -> Self;
+
+    /// Return a darker version of this color. The `amount` should be between 0.0 and 1.0.
+    /// The amount represents an absolute decrease in luminance, and is commutative:
+    /// `color.darker(a).darker(b) == color.darker(a + b)`.
+    ///
+    /// For a relative decrease in luminance, you can simply `mix()` with black.
+    fn darker(&self, amount: f32) -> Self;
+
+    /// Return a lighter version of this color. The `amount` should be between 0.0 and 1.0.
+    /// The amount represents an absolute increase in luminance, and is commutative:
+    /// `color.lighter(a).lighter(b) == color.lighter(a + b)`.
+    ///
+    /// For a relative increase in luminance, you can simply `mix()` with white.
+    fn lighter(&self, amount: f32) -> Self;
+}
+
+/// Linear interpolation of two colors within a given color space.
+pub trait Mix: Sized {
```

Comment:
Blocking concern: we need to figure out how this trait relates to `Animatable`. Our options:

1. Use and implement `Animatable` directly, and axe this trait.
2. Use a blanket `impl <T: Mix> Animatable for T` that acts as a bridging layer.

I think that 1 is clearer and results in less duplication, but will complicate our dependency tree and make this crate effectively useless externally. As a result, I think that 2 is the way to go.

Reply:
3. Move the stuff to a wrapper type, e.g. `PerceputallyLinearMix<T>`. Or differentiate perceptually vs mathematically linear by the underlying color space types.